quickstart/101-ai-studio/dependent.tf

```diff
@@ -0,0 +1,67 @@
+
+resource "azurerm_resource_group" "default" {
+  name     = "azapi-template-rg-${var.names}"
+  location = var.location
+}
+
+resource "azurerm_storage_account" "default" {
+  name                            = "${var.names}storage"
+  location                        = azurerm_resource_group.default.location
+  resource_group_name             = azurerm_resource_group.default.name
+  account_tier                    = "Standard"
+  account_replication_type        = "GRS"
+  allow_nested_items_to_be_public = false
+}
+
+resource "azurerm_key_vault" "default" {
+  name                     = "${var.names}keyvault"
+  location                 = azurerm_resource_group.default.location
+  resource_group_name      = azurerm_resource_group.default.name
+  tenant_id                = data.azurerm_client_config.current.tenant_id
+  sku_name                 = "standard"
+  purge_protection_enabled = false
+}
+
+// AzAPI AIServices
+resource "azapi_resource" "AIServicesResource"{
+  type = "Microsoft.CognitiveServices/accounts@2024-04-01-preview"
```

Comment:
I do not have any major concerns with it but why are we using AzApI here instead of the AzureRM resource?

Reply:
AzureRM does not currently support kind="AIServices", so this is to get around that. Once AIServices is supported, it will be switched to the AzureRM resource.